Mathlib/Data/Set/Finite.lean

```diff
@@ -1353,6 +1353,24 @@ theorem exists_upper_bound_image [Nonempty α] [LinearOrder β] (s : Set α) (f
     (h : s.Finite) : ∃ a : α, ∀ b ∈ s, f b ≤ f a :=
   exists_lower_bound_image (β := βᵒᵈ) s f h
 
+lemma map_finite_iSup {F ι : Type*} [CompleteLattice α] [CompleteLattice β] [FunLike F α β]
+    [SupBotHomClass F α β] [Finite ι] (f : F) (g : ι → α) :
+    f (⨆ i, g i) = ⨆ i, f (g i) := by
+  rw [← iSup_univ (f := g), ← iSup_univ (f := fun i ↦ f (g i))]
+  apply Finite.induction_on (C := fun s ↦ f (⨆ x ∈ s, g x) = ⨆ x ∈ s, f (g x)) finite_univ
```

Comment:
Can you prove `f (⨆ x ∈ s, g x) = ⨆ x ∈ s, f (g x)` first? Also please check if you can use the `induction` tactic (which might be easier in that case too)

Reply:
Done. With this idea, I can actually use `map_finset_sup`, so there is not need for any induction. Thanks!